### ISSUE TITLE
How to set maxAge(duration) of cookie to -1

### ISSUE THREAD

**Issue by kobevaliant** [ktorio/ktor#1367]
Hi,
I want the session deleted when user close browser, so I must set the cookie's maxAge to -1, but I don't know how to do it. The folowing code doesn't work
```
    install(Sessions) {
        cookie<FfpSession>(
            "SESSION_ID",
            directorySessionStorage(File(".sessions"), cached = true)
        ) {
            cookie.path = "/"
//            cookie.duration = Duration.ofSeconds(-1)
        }
    }
```

I find some code in ktor in `io.ktor.sessions.SessionTransportCookie`
```
    override fun send(call: ApplicationCall, value: String) {
        val now = GMTDate()
        val maxAge = configuration.duration?.let { it[ChronoUnit.SECONDS].coerceAtMost(Int.MAX_VALUE.toLong()) }
        val expires = maxAge?.let { now + TimeUnit.SECONDS.toMillis(maxAge) }

        val cookie = Cookie(
            name,
            transformers.transformWrite(value),
            configuration.encoding,
            maxAge?.toInt() ?: 0,
            expires,
            configuration.domain,
            configuration.path,
            configuration.secure,
            configuration.httpOnly,
            configuration.extensions
        )

        call.response.cookies.append(cookie)
    }
```
It seems that `expires` is set without considerring maxAge of -1?

**Comment by cy6erGn0m** [comment]
One usually can't deal with negative time periods so nobody knows what is a duration of -1 seconds.

**Comment by cy6erGn0m** [comment]
According to MDN https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Session_cookies

> 

An HTTP cookie (web cookie, browser cookie) is a small piece of data that a server sends to the user's web browser. The browser may store it and send it back with the next request to the same server. Typically, it's used to tell if two requests came from the same browser — keeping a user logged-in, for example. It remembers stateful information for the stateless HTTP protocol.

Cookies are mainly used for three purposes:

Session management
    Logins, shopping carts, game scores, or anything else the server should remember
Personalization
    User preferences, themes, and other settings
Tracking
    Recording and analyzing user behavior

Cookies were once used for general client-side storage. While this was legitimate when they were the only way to store data on the client, it is recommended nowadays to prefer modern storage APIs. Cookies are sent with every request, so they can worsen performance (especially for mobile data connections). Modern APIs for client storage are the Web storage API (localStorage and sessionStorage) and IndexedDB.

To see stored cookies (and other storage that a web page can use), you can enable the Storage Inspector in Developer Tools and select Cookies from the storage tree.
Creating cookies
Section

When receiving an HTTP request, a server can send a Set-Cookie header with the response. The cookie is usually stored by the browser, and then the cookie is sent with requests made to the same server inside a Cookie HTTP header. An expiration date or duration can be specified, after which the cookie is no longer sent. Additionally, restrictions to a specific domain and path can be set, limiting where the cookie is sent.
The Set-Cookie and Cookie headers
Section

The Set-Cookie HTTP response header sends cookies from the server to the user agent. A simple cookie is set like this:

Set-Cookie: <cookie-name>=<cookie-value>

This header from the server tells the client to store a cookie.
Note: Here's how to use the Set-Cookie header in various server-side applications:

    PHP
    Node.JS
    Python
    Ruby on Rails

HTTP/2.0 200 OK
Content-type: text/html
Set-Cookie: yummy_cookie=choco
Set-Cookie: tasty_cookie=strawberry

[page content]

Now, with every new request to the server, the browser will send back all previously stored cookies to the server using the Cookie header.

GET /sample_page.html HTTP/2.0
Host: www.example.org
Cookie: yummy_cookie=choco; tasty_cookie=strawberry

Session cookies
Section

The cookie created above is a session cookie: it is deleted when the client shuts down, because it didn't specify an Expires or Max-Age directive.

So what you need is missing both Expires and MaxAge that you can achieve by setting `duration` to `null`.

**Comment by e5l** [comment]
Released in 1.3.0
